File: micronaut-camunda-example/src/test/java/info/novatec/micronaut/camunda/example/ProcessTest.java
package info.novatec.micronaut.camunda.example;

import io.micronaut.test.annotation.MicronautTest;
import org.camunda.bpm.engine.RuntimeService;
import org.camunda.bpm.engine.delegate.DelegateExecution;
import org.junit.jupiter.api.Test;

import javax.inject.Inject;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.verify;

@MicronautTest
class ProcessTest {

    @Inject
    RuntimeService runtimeService;

    @Inject
    LoggerDelegate loggerDelegate;

    @Test
    public void verifyBeanInvocationInServiceTask() {
        runtimeService.startProcessInstanceByKey("HelloWorld");
        verify(loggerDelegate).execute(any(DelegateExecution.class));
    }
}

File: example-external-task-process/src/test/kotlin/info/novatec/micronaut/camunda/bpm/externaltask/process/ProcessTest.kt
/*
 * Copyright 2021 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package info.novatec.micronaut.camunda.bpm.externaltask.process

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import org.camunda.bpm.engine.RuntimeService
import org.camunda.bpm.engine.test.assertions.bpmn.BpmnAwareTests.*
import org.junit.jupiter.api.Test
import javax.inject.Inject

@MicronautTest
class ProcessTest {

    @Inject
    lateinit var runtimeService: RuntimeService

    @Test
    fun happyPath() {
        val processInstance = runtimeService.startProcessInstanceByKey("externalTaskProcessExample")
        assertThat(processInstance).isStarted
        assertThat(processInstance).isWaitingAt("externalTask").externalTask().hasTopicName("my-topic")
        complete(externalTask(), withVariables("approved", true))
        assertThat(processInstance).hasPassed(
                "createItem",
                "externalTask",
                "logResult"
        )
    }

}